src/mo_frontend/typing.ml

```diff
@@ -3357,6 +3372,16 @@ and infer_dec_valdecs env dec : Scope.t =
     let obj_typ = object_of_scope env obj_sort.it dec_fields obj_scope' at in
     let _ve = check_pat env obj_typ pat in
     Scope.{empty with val_env = singleton id obj_typ}
+  | LetD (pat, exp, None) when is_value_import dec ->
+    let typ, val_env = match recover_opt (infer_pat false {env with msgs = Diag.silenced}) pat with
```

Comment:
I'd avoid backtracking here, also this Diag.silenced would not be necessary.
We basically support two kinds of patterns for these value imports: annotated (Blob or Text) or not which means defaults to Text.
I'd just check if there is an annotation, then call check_pat_exhaustive.
Otherwise call infer_pat_exhaustive.

Reply:
Yeah, I strongly agree.

Reply:
Then all this code can go away and check_import just deals with it.